docs/api/dataset.rst

```diff
@@ -0,0 +1,101 @@
+.. _dataset_api :
+
+Dataset
+=======
+
+Dataset API
+
+
+data_export
+-----------
+
+.. automodule:: qcodes.dataset.data_export
+   :members:
+
+data_set
+--------
+
+.. automodule:: qcodes.dataset.data_set
+   :members:
+
+database
+--------
+
+.. automodule:: qcodes.dataset.database
+   :members:
+
+database_extract_runs
+---------------------
+
+.. automodule:: qcodes.dataset.database_extract_runs
+   :members:
+
+
+dependencies
+------------
+
+.. automodule:: qcodes.dataset.dependencies
+   :members:
+
+descriptions
+------------
+
+.. automodule:: qcodes.dataset.descriptions
+   :members:
+
+experiment_container
+--------------------
+
+.. automodule:: qcodes.dataset.experiment_container
+   :members:
+
+guids
+-----
+
+.. automodule:: qcodes.dataset.guids
+   :members:
+
+json_exporter
+-------------
+
+.. automodule:: qcodes.dataset.json_exporter
+   :members:
+
+legacy_import
+-------------
+
+.. automodule:: qcodes.dataset.legacy_import
+   :members:
+
+
+measurements
+------------
+
+.. automodule:: qcodes.dataset.measurements
+   :members:
+
+
+param_spec
+----------
+
+.. automodule:: qcodes.dataset.param_spec
+   :members:
+
+
+plotting
+--------
+
+.. automodule:: qcodes.dataset.plotting
+   :members:
+
+sqlite_base
```

Comment:
do we really want to generate docs for everything? i thought that for example sqlite_base is something internal that noone should use outside qcodes internals, am i wrong?

Reply:
This is still undecided and why this is marked wip